server/src/main/scala/controllers/application/WithExports.scala

```diff
@@ -86,51 +76,58 @@ trait WithExports extends WithDesksExport with WithFlightsExport {
           now.getLocalNextMidnight
         } else startOfWeekMidnight
 
-        val portStateFuture = ctrl.portStateActor.ask(
-          GetPortStateForTerminal(startOfForecast.millisSinceEpoch, endOfForecast.millisSinceEpoch, terminal)
-          )(new Timeout(30 seconds))
-
+        val portStateFuture = portStateForTerminal(terminal, endOfForecast, startOfForecast)
         val fileName = f"${airportConfig.portCode}-$terminal-forecast-export-headlines-${startOfForecast.getFullYear()}-${startOfForecast.getMonth()}%02d-${startOfForecast.getDate()}%02d"
-        portStateFuture.map {
-          case Some(portState: PortState) =>
-            val hf: ForecastHeadlineFigures = Forecast.headlineFigures(startOfForecast, endOfForecast, terminal, portState, airportConfig.queuesByTerminal(terminal).toList)
-            val csvData = CSVData.forecastHeadlineToCSV(hf, airportConfig.exportQueueOrder)
-            Result(
-              ResponseHeader(200, Map("Content-Disposition" -> s"attachment; filename=$fileName.csv")),
-              HttpEntity.Strict(ByteString(csvData), Option("application/csv")
-                                )
-              )
-
-          case None =>
-            log.error(s"Missing headline data for ${startOfWeekMidnight.ddMMyyString} for Terminal $terminal")
-            NotFound(s"Sorry, no headlines available for week starting ${startOfWeekMidnight.ddMMyyString}")
+
+        portStateFuture.map { portState =>
+          val hf: ForecastHeadlineFigures = Forecast.headlineFigures(startOfForecast, endOfForecast, terminal, portState, airportConfig.queuesByTerminal(terminal).toList)
+          val csvData = CSVData.forecastHeadlineToCSV(hf, airportConfig.exportQueueOrder)
+          csvFileResult(fileName, csvData)
         }
       }
     }
   }
 
-  def queryPortStateActor: (SDateLike, Any) => Future[Option[PortState]] = (from: SDateLike, message: Any) => {
+  def csvFileResult(fileName: String, data: String): Result = Result(
+    ResponseHeader(200, Map("Content-Disposition" -> s"attachment; filename=$fileName.csv")),
+    HttpEntity.Strict(ByteString(data), Option("application/csv")))
+
+  def portStateForTerminal(terminal: Terminal,
+                           endOfForecast: SDateLike,
+                           startOfForecast: SDateLike): Future[PortState] =
+    ctrl.portStateActor
+      .ask(GetPortStateForTerminal(startOfForecast.millisSinceEpoch, endOfForecast.millisSinceEpoch, terminal))(new Timeout(30 seconds))
+      .mapTo[PortState]
+      .recover {
+        case t =>
+          log.error("Failed to get PortState", t)
+          PortState.empty
+      }
+
+  def queryFromPortState: (SDateLike, Any) => Future[Any] = (from: SDateLike, message: Any) => {
```

Comment:
isn't this the same as saying

```
queryFromPortState(from: SDateLike, message: Any): (SDateLike, Any) => Future[Any] = {
...
```

Reply:
Almost. It's the similar to:
```queryFromPortState(from: SDateLike, message: Any): Future[Any] = {```

But I'm wanting to pass it as a function, so I should have made it a val, and I've also changed the name to make it more obvious it's a function:
```val queryFromPortStateFn: (SDateLike, Any) => Future[Any] = (from: SDateLike, message: Any) => {```